docs/snippets/ShapeInference_1.cpp

```diff
@@ -0,0 +1,36 @@
+#include <openvino/runtime/core.hpp>
+
+
+int main() {
+std::string device = "CPU";
+//! [part0]
+    ov::Core core;
+    // ------------- 0. Read model ------------------------------------------------------
+    auto model = core.read_model("model_file");
+    // ------------- 2. Collect original map of input index/name/object to its shape ----
+    // this is an example on how to collect all three maps, in the application use one
+    std::map<ov::Output<ov::Node>, ov::PartialShape> obj_to_shape;
+    size_t i = 0;
+    std::map<size_t, ov::PartialShape> idx_to_shape;
+    std::map<std::string, ov::PartialShape> name_to_shape;
+    for (const ov::Output<ov::Node>& input : model->inputs()) {
+        ov::PartialShape shape = input.get_partial_shape();
+        obj_to_shape[input] = shape;
+        idx_to_shape[i++] = shape;
+        // input may have no name, in such case use map based on input index or object instead
+        if (!input.get_names().empty())
+            name_to_shape[input.get_any_name()] = shape;
+    }
+    // ------------- 3. Modify shapes in the map to reflect desired input shapes ---------
+    // ...
+    // ------------- 4. Call reshape to adjust models input ------------------------------
+    // this is an example on how to call method for all three maps, in the application use
+    // one reshape call per single shape adjustment
+    model->reshape(obj_to_shape);
+    model->reshape(idx_to_shape);
+    model->reshape(name_to_shape);
+    // ------------- 5. Compile model with updated shapes for the device -----------------
+    auto compiled_model = core.compile_model(model, device);
+//! [part0]
```

Comment:
can / should we have code snippets with tabs for `Tensor name`, `Index`,  `Port` and user can select between them?
Now, the example looks complex because it mixes several types of reshape. I would prefer a single snippet for a single usecase

Secondly, do we need to demonstare e2e (starting from read ending with compile_model?) or only reshape functionality. In this case snippets will be more compact and easy-readble, while now half of the snippets - is not a topic for current guide and it has a lot of comments.

This is true for all snippets.